generators/client/templates/angular/src/main/webapp/app/shared/auth/_authorities-constants.ts

```diff
@@ -0,0 +1,7 @@
+export class AuthoritiesConstants {
```

Comment:
Regarding singular, I kept the same name as server side (see generators\server\templates\src\main\java\package\security\_AuthoritiesConstants.java). 
Regarding enum suport, unfortunately for typescript version <2.4 enums do not have string literals. 
In 2.4 are/will be supported, [see blog announcement](https://blogs.msdn.microsoft.com/typescript/2017/06/12/announcing-typescript-2-4-rc/#user-content-string-enums), but I guess will have to wait for the version upgrade.